DNN Platform/Library/Common/Initialize.cs

```diff
@@ -428,41 +428,46 @@ private static string InitializeApp(HttpApplication app, ref bool initialized)
             {
                 // Check whether the current App Version is the same as the DB Version
                 redirect = CheckVersion(app);
-                if (string.IsNullOrEmpty(redirect) && !InstallBlocker.Instance.IsInstallInProgress())
+                if (!string.IsNullOrEmpty(redirect) || InstallBlocker.Instance.IsInstallInProgress())

```

Comment:
Since this condition was reversed, would it make sense to bring all the core from the else block up here and kill the else block ?

Reply:
I finished simplifying this logic, let me know if you were looking for anything different.